ompi/errhandler/errhandler_invoke.c

```diff
@@ -48,7 +48,22 @@ int ompi_errhandler_invoke(ompi_errhandler_t *errhandler, void *mpi_object,
         if (state >= OMPI_MPI_STATE_INIT_COMPLETED &&
             state < OMPI_MPI_STATE_FINALIZE_PAST_COMM_SELF_DESTRUCT) {
             comm = (ompi_mpi_compat_mpi3)? &ompi_mpi_comm_world.comm: &ompi_mpi_comm_self.comm;
-            comm->error_handler->eh_comm_fn(&comm, &err_code, message, NULL);
+            switch (comm->error_handler->eh_lang) {
+               case OMPI_ERRHANDLER_LANG_C:
+                 comm->error_handler->eh_comm_fn(&comm, &err_code, message, NULL);
+                 break;
+
+               case OMPI_ERRHANDLER_LANG_CXX:
```

Comment:
Open MPI doesn't have the C++ bindings any more -- should we remove all references to `OMPI_ERRHANDLER_LANG_CXX`?

Reply:
If we decide to keep the internal C++ support we should at least pass the C++ error handler and not the C one as the last parameter of the callback.

Reply:
Good point.

I think we should probably just ditch all the C++ glue here -- `OMPI_ERRHANDLER_LANG_CXX`, the `eh_cxx_dispatch_fn`, ... etc.  It's all dead code nowadays.